pkgs/applications/emulators/86box/default.nix

```diff
@@ -48,14 +49,26 @@ stdenv.mkDerivation rec {
     ++ lib.optional (!enableDynarec) "-DDYNAREC=OFF"
     ++ lib.optional (!unfreeEnableDiscord) "-DDISCORD=OFF";
 
-  postInstall = lib.optional stdenv.isLinux ''
+  roms = lib.optionalString unfreeEnableRoms (fetchFromGitHub {
+    owner = "86Box";
+    repo = "roms";
+    rev = "v${version}";
+    hash = "sha256-1HtoizO0QIGNjQTW0clzRp40h1ulw55+iTYz12UJSms=";
+  });
+
+  postInstall = ''
+    if [ ! -z "$roms" ]; then
```

Comment:
This is terrible. Why not using Nix to perform such a check? How about using a `let in` predicate in the beginning of the expression, and set `roms` there unconditionally, and then in the `postInstall`, use `lib.optionalString unfreeEnableRoms`?

Reply:
It uses the version, I plan to make it compatible with r-ryantm.